Make Ops::ChargeForward() take in input sizes parameter

evsmithx opened the pull request:
Ops::ChargeForward() now takes a vector of input sizes as a parameter, and returns the output size as well as the charge.

evsmithx committed:
Make ChargeForward take in input sizes

baykaner commented on libs/ml/include/ml/ops/ops.hpp:
```diff
@@ -170,14 +170,28 @@ class Ops
+    if (!input_shapes.empty()) {
+      output_shape.back() = input_shapes.back().back();  // let's hope this is the batch size...
+    }
+    return std::make_pair(0, output_shape);
```
NEVER RETURN 0 !!!!!

evsmithx committed:
Merging

MissingNO57 committed:
Coefs fine tuning (#2437)

Weights initialization happens on Compile and not on LayerAdd
BM_ForwardAdd benchmarks improved
Ops::Add::Forward charge computation improved
Added Tensor iteration charge estimate

evsmithx committed:
Adding ComputeShape that takes sizevectors

evsmithx committed:
Merging
